client/src/apollo/apollo-client-hooks.jest.spec.js

```diff
@@ -0,0 +1,195 @@
+        };
+
+        apolloMock.query.mockResolvedValue({
+            data: {
+                currentUser: null
+            }
+        });
+
+        const next = jest.fn();
+
+        await beforeEachRequiresAuth(apolloMock, to, undefined, next);
+
+        expect(apolloMock.query).toHaveBeenCalled();
+        expect(setSession).toHaveBeenCalled();
+        expect(next).toHaveBeenCalled();
+        expect(next.mock.calls[0][0]).not.toBe(undefined);
+    });
+
+    it('does nothing it route does not require auth', async () => {
+        const to = {
+            matched: [{ meta: {} }]
+        };
+
+        const next = jest.fn();
+
+        await beforeEachRequiresAuth(apolloMock, to, undefined, next);
+
+        expect(apolloMock.query).not.toHaveBeenCalled();
+        expect(next).toHaveBeenCalled();
+        expect(next.mock.calls[0][0]).toBe(undefined);
+    });
+});
+
+describe('Router RequireRoles hook', () => {
+    afterEach(() => {
+        apolloMock.query.mockClear();
+    })
+
+    it('does nothing when user has role', async () => {
+        const to = {
+            matched: [{ meta: { requiresRoles: 'admin' } }]
+        };
+
+        apolloMock.query.mockResolvedValue({
+            data: {
+                currentUser: {
+                    id: 1,
+                    roles: [
+                        { name: 'admin'}
+                    ]
+                }
+            }
+        });
+
+        const next = jest.fn();
+
+        await beforeEachRequiresRoles(apolloMock, to, undefined, next);
+
+        expect(apolloMock.query).toHaveBeenCalled();
+        expect(next).toHaveBeenCalled();
+        expect(next.mock.calls[0][0]).toBe(undefined);
+    });
+
+    it('redirects when user does not have role', async () => {
+        const to = {
+            matched: [{ meta: { requiresRoles: 'admin' } }]
+        };
+
+        apolloMock.query.mockResolvedValue({
+            data: {
+                currentUser: {
+                    id: 1,
+                    roles: [
+
+                    ]
+                }
+            }
+        });
+
+        const next = jest.fn();
+
+        await beforeEachRequiresRoles(apolloMock, to, undefined, next);
+
+        expect(apolloMock.query).toHaveBeenCalled();
+        expect(next).toHaveBeenCalled();
+        expect(next.mock.calls[0][0]).not.toBe(undefined);
+    });
+
+    it('does nothing is requireRoles is not present', async () => {
+        const to = {
+            matched: [{ meta: {} }]
+        };
+
+        const next = jest.fn();
+
+        await beforeEachRequiresRoles(apolloMock, to, undefined, next);
+
+        expect(apolloMock.query).not.toHaveBeenCalled();
+        expect(next).toHaveBeenCalled();
+        expect(next.mock.calls[0][0]).toBe(undefined);
+    });
+
+    it('requires all nested roles', async () => {
+        const to = {
+            matched: [{ meta: { requiresRoles: 'admin' } }, {meta: { requiresRoles: 'specialAdmin' } }]
```

Comment:
What does a "specialAdmin" represent?

Reply:
a random role that's not really a thing.

Reply:
What do you think about renaming this `specialTestAdmin` just to be clear about the purpose of this object?

Reply:
I changed the name in the latest commit.